docs/aws/audit/kmsmonitoring/rules/secrets_manager_in_use.mdx

```diff
@@ -23,6 +23,258 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access KMS and Secrets Manager.
+
+2. Once AWS CLI is installed and configured, you can list all the keys in your account by running the following command:
+
+   ```
+   aws kms list-keys
+   ```
+   This command will return a list of all the keys in your account, including their key IDs and ARNs.
+
+3. To check if Secrets Manager is using these keys, you can list all the secrets in your account by running the following command:
+
+   ```
+   aws secretsmanager list-secrets
+   ```
+   This command will return a list of all the secrets in your account, including their names and ARNs.
+
+4. For each secret, you can describe the secret to see which KMS key it's using by running the following command:
+
+   ```
+   aws secretsmanager describe-secret --secret-id <secret-name>
+   ```
+   Replace `<secret-name>` with the name of the secret. This command will return information about the secret, including the KMS key that it's using. If the `KmsKeyId` field is not present or does not match any of the keys listed in step 2, then Secrets Manager is not using KMS.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you can start writing the script, you need to install the necessary Python libraries. AWS SDK for Python (Boto3) is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python that allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+
+   ```bash
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and create a session: In your Python script, you need to import the necessary libraries and create a session using your AWS credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='us-west-2'
+   )
+   ```
+
+3. List all the keys in AWS KMS: Now, you can use the `list_keys` method to get a list of all the keys in AWS KMS.
+
+   ```python
+   kms = session.client('kms')
+   response = kms.list_keys()
+
+   for key in response['Keys']:
+       print(key['KeyId'])
+   ```
+
+4. Check if Secrets Manager is in use: To check if Secrets Manager is in use, you can list all the secrets and check if the KMS key is used by any of them.
+
+   ```python
+   secrets_manager = session.client('secretsmanager')
+   response = secrets_manager.list_secrets()
+
+   for secret in response['SecretList']:
+       if 'KmsKeyId' in secret and secret['KmsKeyId'] == 'arn:aws:kms:us-west-2:123456789012:key/abcd1234-a123-456a-a12b-a123b4cd56ef':
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>